department-of-veterans-affairs/va.gov-cms
# VACMS-7552 update theme build process

## Description

Closes #7552. Apologies for so many files changed, it's updating both custom admin themes and some build process files. Most relevant files: `gulpfile.es6.js`, `composer.json`, `.tugboat/config.yml`
- Deletes both theme's css assets directory in favor of compiling them to a .gitignored `dist/` directory in the respective themes. This solves the issue of conflicts created every time styles are updated in a PR due to the minified version conflicting. Conflicts may still arise if editing the same source files, as they should.
- Updated all references in both themes to use the new dist/ directory. Includes `*.libraries.yml` and .scss image paths
- Adds scripts for compiling & watching theme to composer.json, removed CMS-theme specific bits from va:web:install
- Adds call to va:theme:compile to va:start and the tugboat.yml files

## Testing done
- [x] verified styles were compiled for vagovadmin and vagovclaro when rebuilding the site locally
- [x] verified styles for both themes are present on tugboat
- [x] verified watch command successfully sees style changes & recompiles & clears drupal caches

## Screenshots
n/a

## QA steps

testing the build command:
- login as admin
- visit PR environment, check centralized content type for example, see that styles are present in vagovadmin theme
- switch theme to vagovclaro, see that styles are present when changed

If styles aren't present, or look a little funky, the compile command did not run as expected. There should be output in the logs with core .pcss files being processed into .css files. This is a required step for vagovclaro to leverage the css variables.

testing the watch command locally:
- Enable vagovclaro as the admin theme
- run `lando composer va:theme:compile` (or the ddev equivalent). A gulp task should start.
- Edit `docroot/themes/custom/vagovclaro/assets/scss/components/_layout.scss`, change the content-header background color. The gulp task should recompile sass and clear drupal cache
- Refresh your page, see the updated color.

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
